src/D2L.CodeStyle.Analyzers/ApiUsage/DependencyInjection/DependencyRegistrationsAnalyzer.cs

```diff
@@ -106,21 +109,27 @@ out dependencyRegistrationExpresion
 				return;
 			}
 
-			InspectRegistration( dependencyRegistration, context );
+			InspectRegistration( dependencyRegistration, immutabilityCtx, context );
 		}
 
-		private void InspectRegistration( DependencyRegistration registration, SyntaxNodeAnalysisContext ctx ) {
+		private void InspectRegistration(
+			DependencyRegistration registration,
+			ImmutabilityContext immutabilityCtx,
+			SyntaxNodeAnalysisContext ctx
+		) {
 			if( registration.ObjectScope == ObjectScope.Singleton ) {
 				var typesToInspect = GetTypesRequiredToBeImmutableForSingletonRegistration( registration );
 				foreach( var type in typesToInspect ) {
+					if( type.IsNullOrErrorType() ) {
+						continue;
+					}
 
 					// We require full immutability here,
 					// because we don't know if it's a concrete type
 					//
 					// TODO: could make this better by exposing the minimum
 					// scope required for each type
-					var immutabilityScope = type.GetImmutabilityScope();
-					if( !type.IsNullOrErrorType() && immutabilityScope != ImmutabilityScope.SelfAndChildren ) {
+					if( !immutabilityCtx.IsImmutable( type, ImmutableTypeKind.Total, () => null, out var _ ) ) {

```

Comment:
The purpose of this PR is to remove the call to `GetImmutabilityScope()` and usage of `ImmutabilityScope` (which is similar to `ImmutabilityKind` but not `[Flags]` and with weird terminology... I forgot that it existed honestly (it's hidden inside the MutabilityInspector).